R/insertPackage.R

```diff
@@ -220,41 +245,20 @@ getPackageInfo <- function(file) {
     return(fields)
 }
 
-##' This function returns the directory path (relative to
-##' the repo root) where the package needs to be copied to.
-##'
-##' @title Get relative path for package type
-##' @param file The fully qualified path of the package
-##' @return string Relative file path where packages of
-##' this type should be copied to.
-##' @author Jan Schulz, Dirk Eddelbuettel and Matthew Jones
-getPathForPackage <- function(file) {
-    pkgtype <- identifyPackageType(file)
-    fields <- getPackageInfo(file)
-    rversion <- unname(fields["Rmajor"])
-
-    if (pkgtype == "source") {
-        ret <- file.path("src", "contrib")
-    } else if (pkgtype == "win.binary") {
-        ret <- file.path("bin", "windows", "contrib", rversion)
-    } else if (pkgtype == "mac.binary") {
-        if (unname(fields["OSflavour"]) == "") {
-            # non-binary package, treated as el-capitan
-            if (grepl("mac.binary", .Platform$pkgType)) {
-              fields["osxFolder"] <- gsub("mac.binary.", "", .Platform$pkgType)
-            } else {
-              fields["osxFolder"] <- "el-capitan"
-            }
-            message("Note: Non-binary OS X package will be installed in ", fields["osxFolder"], " path.")
-        }
-        if (unname(fields["osxFolder"]) != "") {
-            ret <- file.path("bin", "macosx", fields["osxFolder"], "contrib", rversion)
-        } else {
-            ret <- file.path("bin", "macosx", "contrib", rversion)
-        }
+contrib.url2 <- function(repos, type = getOption("pkgType"), version = NULL){
+    contrib_url <- contrib.url(repos = repos, type = type)
+    if(is.null(version) || is.na(version)){
+        return(contrib_url)
     }
-    return(ret)
+    version <- as.character(version)
+    split_version <- strsplit(version,"\\.")[[1L]]
+    if(length(split_version) < 2L || length(split_version) > 3L){
+        stop("'version' must be in the format 'X.Y' or 'X.Y.Z'.")
+    }
+    version <- paste(split_version[seq.int(1L,min(2L,length(split_version)))],
+                     collapse = ".")
```

Comment:
If you convert version to a `base::package_version()` instead of `as.character`, you can replace all the string munging with

    paste0(version$major, '.', version$minor)

Reply:
I thought about this, but there is no base R function I could find, which reports `version$minor` as `6` (for R `3.6`). Therefore the default output is `getRversion()`, which returns `3.6.3` a special class, which is not `character` and thus has to be converted.

Reply:
I missed that. I use that a lot too (somewhere even from `configure`.  That S3 class `package_version` and its constructors, and added operators, is gold.  Hidden jewel.

Reply:
I am happy to change that, if there is a function, which reports `3.6` or `4.0` and not `3.6.3` or `4.0.0`

Reply:
Ok, now I got it. Give me a moment

Reply:
I was thinking something like this:

```
> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/atlas/libblas.so.3.10.3
LAPACK: /usr/lib/x86_64-linux-gnu/atlas/liblapack.so.3.10.3

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] colorout_1.2-2

loaded via a namespace (and not attached):
[1] compiler_3.6.3 tools_3.6.3   
> version <- getRversion()
> version$minor
[1] 6
> paste0(version$major, '.', version$minor)
[1] "3.6"

```

Reply:
done. I didn't realize, that there are these accessor, even though there are referenced in the examples.

Thanks for the checking that.

Reply:
Yes, high redundancy.  getRversion() and alike give us the components.  The S3 class gives us comparison.